syntax = "proto3";

package proto;

import "proto/common.proto";


message BlockHeader {
	
	string block_hash = 1;
	uint64 block_number = 2;
	int64 timestamp = 3;
	string previous_block_hash = 4;
	string merkle_root = 5;
	repeated string merkle_proof = 6;
	repeated string transaction_hashes =  7; // optional for debugging and verification

}

message FileAuditResponse {
	
	string merkle_root = 1;
	repeated string merkle_proof = 2 ;
	int64 audit_index = 3 ;
	string status = 4 ;  // success, failure
	string error_message = 5; // If its failure

}

service FileAuditService {
	rpc SubmitAudit ( proto.FileAuditRequest ) returns ( FileAuditResponse );
}



